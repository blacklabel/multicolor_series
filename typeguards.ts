import { Series, Point } from 'highcharts';
import { ColoredlineSeries, ColoredareaSeries } from './ts/MulticolorSeries';
import { SeriesColoredPoint, SeriesColoredSegment } from './types';

type ExtendedColoredlineSeries = ColoredlineSeries & {
    segments: SeriesColoredSegment<SeriesColoredPoint<Point>>[];
};

type ExtendedColoredareaSeries = ColoredareaSeries & {
    segments: SeriesColoredSegment<SeriesColoredPoint<Point>>[];
};

export const isSeriesColored = (
    series: Series | ExtendedColoredlineSeries | ExtendedColoredareaSeries
): series is ExtendedColoredlineSeries | ExtendedColoredareaSeries => true;
