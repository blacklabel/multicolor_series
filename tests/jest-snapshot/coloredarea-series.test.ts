import Highcharts from 'highcharts';
import multicolorModule from '../../dist/multicolor-series';
import { isSeriesColored } from './../../typeguards';
import { generateFormattedSegments, generateFormattedSeries } from './helper';

multicolorModule(Highcharts);

describe('Refactoring regression tests - series coloredarea.', (): void => {
    const containerElement = document.createElement('div');
    const chart = Highcharts.chart(containerElement, {
        chart: {
            animation: false
        },

        plotOptions: {
            series: {
                animation: false
            }
        },

        accessibility: {
            enabled: false
        },

        series: [
            {
                type: 'coloredarea',
                data: [
                    {
                        y: 40,
                        segmentColor: 'red'
                    },
                    {
                        y: 60,
                        segmentColor: 'blue'
                    },
                    {
                        y: 30,
                        segmentColor: 'yellow'
                    },
                    {
                        y: 10,
                        segmentColor: 'green'
                    },
                    {
                        y: 50,
                        segmentColor: 'green'
                    },
                    {
                        y: 20,
                        segmentColor: 'brown'
                    },
                    {
                        y: 70,
                        segmentColor: 'pink'
                    }
                ]
            }
        ]
    });

    const series = chart.series[0];

    if (!series) {
        throw Error('Series should be defined.');
    }

    if (!isSeriesColored(series)) {
        throw Error('Series type should be coloredarea.');
    }

    const graphs = series.graphs;

    describe('Graphs element tests.', (): void => {
        test(
            'The graphs element should be an five elements array.',
            (): void => {
                expect(graphs.length).toEqual(5);
            }
        );

        test('The graphs paths should match the snapshot.', (): void => {
            const paths = graphs.map((graph): string =>
                graph.element.outerHTML
            );
            expect(paths).toMatchSnapshot();
        });
    });

    test('The tracker element should match the snapshot.', (): void => {
        expect(series?.tracker?.element).toMatchSnapshot();
    });

    test('The series data should match the snapshot.', (): void => {
        const formattedSeries = generateFormattedSeries(series, series.data);
        expect(formattedSeries).toMatchSnapshot();
    });

    test('The segments data should match the snapshot.', (): void => {
        const formattedSegments = generateFormattedSegments(series.segments);
        expect(formattedSegments).toMatchSnapshot();
    });
});
