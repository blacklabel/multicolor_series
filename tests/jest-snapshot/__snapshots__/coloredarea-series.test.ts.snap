// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Refactoring regression tests - series coloredarea. Graphs element tests. The graphs paths should match the snapshot. 1`] = `
[
  "<path fill="red" d="M 0 0" stroke="red" stroke-width="2" data-z-index="1" stroke-linejoin="round" stroke-linecap="round" filter="none"></path>",
  "<path fill="blue" d="M 0 0" stroke="blue" stroke-width="2" data-z-index="1" stroke-linejoin="round" stroke-linecap="round" filter="none"></path>",
  "<path fill="yellow" d="M 0 0" stroke="yellow" stroke-width="2" data-z-index="1" stroke-linejoin="round" stroke-linecap="round" filter="none"></path>",
  "<path fill="green" d="M 0 0" stroke="green" stroke-width="2" data-z-index="1" stroke-linejoin="round" stroke-linecap="round" filter="none"></path>",
  "<path fill="brown" d="M 0 0" stroke="brown" stroke-width="2" data-z-index="1" stroke-linejoin="round" stroke-linecap="round" filter="none"></path>",
]
`;

exports[`Refactoring regression tests - series coloredarea. The segments data should match the snapshot. 1`] = `
[
  {
    "color": "red",
    "points": [
      {
        "segmentColor": "red",
        "x": 0,
        "y": 40,
      },
      {
        "segmentColor": "blue",
        "x": 1,
        "y": 60,
      },
    ],
  },
  {
    "color": "blue",
    "points": [
      {
        "segmentColor": "blue",
        "x": 1,
        "y": 60,
      },
      {
        "segmentColor": "yellow",
        "x": 2,
        "y": 30,
      },
    ],
  },
  {
    "color": "yellow",
    "points": [
      {
        "segmentColor": "yellow",
        "x": 2,
        "y": 30,
      },
      {
        "segmentColor": "green",
        "x": 3,
        "y": 10,
      },
    ],
  },
  {
    "color": "green",
    "points": [
      {
        "segmentColor": "green",
        "x": 3,
        "y": 10,
      },
      {
        "segmentColor": "green",
        "x": 4,
        "y": 50,
      },
      {
        "segmentColor": "brown",
        "x": 5,
        "y": 20,
      },
    ],
  },
  {
    "color": "brown",
    "points": [
      {
        "segmentColor": "brown",
        "x": 5,
        "y": 20,
      },
      {
        "segmentColor": "pink",
        "x": 6,
        "y": 70,
      },
    ],
  },
]
`;

exports[`Refactoring regression tests - series coloredarea. The series data should match the snapshot. 1`] = `
{
  "data": [
    {
      "segmentColor": "red",
      "x": 0,
      "y": 40,
    },
    {
      "segmentColor": "blue",
      "x": 1,
      "y": 60,
    },
    {
      "segmentColor": "yellow",
      "x": 2,
      "y": 30,
    },
    {
      "segmentColor": "green",
      "x": 3,
      "y": 10,
    },
    {
      "segmentColor": "green",
      "x": 4,
      "y": 50,
    },
    {
      "segmentColor": "brown",
      "x": 5,
      "y": 20,
    },
    {
      "segmentColor": "pink",
      "x": 6,
      "y": 70,
    },
  ],
  "seriesType": "coloredarea",
}
`;

exports[`Refactoring regression tests - series coloredarea. The tracker element should match the snapshot. 1`] = `
<path
  class="highcharts-tracker"
  d="M 0 0"
  data-z-index="2"
  fill="none"
  stroke="rgba(192,192,192,0.002)"
  stroke-linejoin="round"
  stroke-width="2"
  visibility="visible"
/>
`;
